### PR TITLE
Fix var, param and return tag type hint format

### PR DESCRIPTION
**Pull request by othercorey**
Refs https://github.com/cakephp/cakephp/issues/15528

This orders the type hints in tags according to the above RFC.

### CODE REVIEW

**Comment by othercorey** [comment]
This is an example of `\Closure` sorting. Not sure what we prefer:
```
-   * @param \Closure|\Psr\Http\Server\MiddlewareInterface|string $middleware Middleware.
+   * @param \Psr\Http\Server\MiddlewareInterface|\Closure|string $middleware Middleware.
```

**Comment by ADmad** [comment]
I thought we would be doing this sorting only for 5.0.

**Comment by othercorey** [comment]
> I thought we would be doing this sorting only for 5.0.

I manually applied it to 4.next to avoid annoying merge conflicts later. I did for the other type hint cleanup.

**Comment by othercorey** [comment]
Another thing to note, the array version of each type is grouped with that type:

`string[]|string|int[]|int`

We discussed converting all array versions to generics `array<string>` which would group all arrays together:

`array<string>|array<int>|string|int`

**Comment by othercorey** [comment]
Updated to include var, psalm- and phpstan- tags.

File: src/I18n/TranslatorRegistry.php [hunk 1]
```diff
@@ -87,7 +87,7 @@ class TranslatorRegistry
      * A CacheEngine object that is used to remember translator across
      * requests.
      *
-     * @var (\Psr\SimpleCache\CacheInterface&\Cake\Cache\CacheEngineInterface)|null
+     * @var (\Psr\SimpleCache\CacheInterface & \Cake\Cache\CacheEngineInterface)|null
```
Comment (othercorey) [inline src/I18n/TranslatorRegistry.php:90]: Looks like have to remove the spacing from intersection types. However, the space does make it a lot easier to read.

**Comment by othercorey** [comment]
Updated to include `type[]` to `array<type>` array type hint conversion. The array hints are left in the original order they were encountered but grouped together with `array`.

**Pull request merged**